docs/spec/directives.rst

```diff
@@ -86,12 +86,6 @@ At runtime a cast always returns the
 expression unchanged -- it does not check the type, and it does not
 convert or coerce the value.
 
-Casts differ from type comments (see the previous section).  When using
```

Comment:
We can also say that `assert_type()` checks that two types are equivalent (this should be at the top of this file, can't comment there). If you prefer, we can keep this for another PR though.

Reply:
> We can also say that `assert_type()` checks that two types are equivalent

That would imply that the following test should pass:

```py
from typing_extensions import assert_type
from typing import Protocol

class Foo(Protocol):
    x: str
    
class Bar(Protocol):
    x: str

def test(x: Foo):
    assert_type(x, Bar)
```

Mypy [agrees](https://mypy-play.net/?mypy=latest&python=3.12&gist=dcae8228f45fe3c23d95775f5cc8ce0a) that this test passes. Pyright [does not](https://pyright-play.net/?pyrightVersion=1.1.361&strict=true&code=GYJw9gtgBALgngBwJYDsDmB9ApgDxllAZyTCKiQgTBBigENDCsaN4EsAoUSWRVNcpWq0ACuBhgAxmAA2HDpJkNCUAGJgwACjFgJ0mQEoAXByhmoOI1EIwQp8wqWMoAIToht4qbOP2zl61t5ABMsYFgsG00A9TBfc3pGZhhWRCxogBpXdwMgA). Pyre reports no errors on the snippet, but it doesn't look like Pyre supports `assert_type` (not sure).

I think I agree that this test should pass, so perhaps this is simply a bug in pyright's handling of `assert_type`.

Reply:
Thanks. The fact that type checkers disagree is probably an indication that we should keep this out of the current PR, it makes enough changes already.